kivy/core/window/__init__.py

```diff
@@ -2017,9 +2051,37 @@ def on_dropfile(self, filename):
             (ios, android etc.)
 
         .. versionadded:: 1.2.0
+
+        .. versionchanged:: 2.1.0
+            Renamed from `on_dropfile` to `on_drop_file`.
         '''
+        if self.get_property_observers('on_dropfile'):
+            self.dispatch('on_dropfile', filename)
```

Comment:
`get_property_observers` is not particularly efficient. I don't think it's worth checking if anything is bound and instead always dispatch.

I also don't like when widgets do any work in `on_xxx` methods, except in user code. Because it's possible someone will inherit the class and overwrite the method and not call `super` - we have not trained our users to call `super` in these callbacks. It's less likely to be a problem, but still.

A better solution is to do the following in `__init__`: `self.fbind('on_drop_file', self.dispatch, 'on_dropfile')` and it should work,

Reply:
`self.fbind('on_drop_file', self.dispatch, 'on_dropfile')` doesn't work because `on_drop_file` passes a window instance, but wrapping with `lambda` does.